src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp

```diff
@@ -980,6 +984,12 @@ bool EDP::valid_matching(
         EPROSIMA_LOG_WARNING(RTPS_EDP, "Incompatible liveliness kinds: offered kind is < than requested kind");
         incompatible_qos.set(fastdds::dds::LIVELINESS_QOS_POLICY_ID);
     }
+    // DataRepresentationQosPolicy
```

Comment:
It looks like this whole method could be replaced by 
```
    return valid_matching(wdata, rdata, reason, incompatible_qos);
```

Reply:
Done in 9a5c3431d